### PR TITLE
OPENEUROPA-2795: Test we can receive translation updates while a job is active

### PR DESCRIPTION
**Pull request by imanoleguskiza**
### Description

We need to be able to receive and update translations while a job is active. 
Add tests and the possibility to manually test this functionality.
### Change log

- Added: Tests for updating translations and new option to mock it.
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: modules/oe_translation_poetry/modules/oe_translation_poetry_mock/src/EventSubscriber/ContentTranslationOverviewAlterSubscriber.php [hunk 1]
```diff
@@ -135,6 +136,16 @@ public function alterOverview(ContentTranslationOverviewAlterEvent $event): void
           ),
         ];
       }
+      if (isset($translated_jobs[$language->getId()])) {
+        $links['update_translation'] = [
```
Comment (dxvargas) [inline modules/oe_translation_poetry/modules/oe_translation_poetry_mock/src/EventSubscriber/ContentTranslationOverviewAlterSubscriber.php]: I would avoid calling this action update. We already have a "translation update" that has nothing to do with this operation. I suggest translation override.

File: modules/oe_translation_poetry/modules/oe_translation_poetry_mock/src/Controller/PoetryMockController.php [hunk 2]
```diff
@@ -203,7 +203,13 @@ public function sendTranslationNotification(JobInterface $tmgmt_job): Response {
     $item = reset($items);
     $data = \Drupal::service('tmgmt.data')->filterTranslatable($item->getData());
     foreach ($data as $field => &$info) {
-      $info['#text'] .= ' - ' . $tmgmt_job->getTargetLangcode();
+      if (isset($info['#translation'])) {
+        $info['#text'] = $info['#translation']['#text'] . ' UPDATE';
```
Comment (dxvargas) [inline modules/oe_translation_poetry/modules/oe_translation_poetry_mock/src/Controller/PoetryMockController.php]: We already have a "translation update" that has nothing to do with this operation.
I suggest here ' OVERRIDDEN'.

File: modules/oe_translation_poetry/modules/oe_translation_poetry_mock/src/EventSubscriber/ContentTranslationOverviewAlterSubscriber.php (hunk 1 above)
Comment (imanoleguskiza) [reply modules/oe_translation_poetry/modules/oe_translation_poetry_mock/src/EventSubscriber/ContentTranslationOverviewAlterSubscriber.php]: I'm ok with calling the link "translation_override" but I think the title needs to be more descriptive about what we are doing. Something like "Receive translation update (mock)" would do it I think?

Comment (dxvargas) [reply modules/oe_translation_poetry/modules/oe_translation_poetry_mock/src/EventSubscriber/ContentTranslationOverviewAlterSubscriber.php]: So we have already
* Accept job (mock)
* Translate job (mock)
* Cancel job (mock)
What about having now one of
* Translate job (mock)
* Translate job again (mock)
* Retranslate job (mock)

**Pull request merged**